api/v1alpha1/ingressnodefirewall_types.go

```diff
@@ -27,50 +27,49 @@ type IngressNodeFirewallICMPRule struct {
 	// +kubebuilder:validation:Maximum:=255
 	// +kubebuilder:validation:Minimum:=0
 	// +optional
-	ICMPType uint8 `json:"icmpType"`
+	ICMPType uint8 `json:"icmpType,omitempty"`
 
 	// ICMPCode define ICMP Code ID (RFC 792).
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Maximum:=16
 	// +kubebuilder:validation:Minimum:=0
 	// +optional
-	ICMPCode uint8 `json:"icmpCode"`
+	ICMPCode uint8 `json:"icmpCode,omitempty"`
 }
 
 // IngressNodeFirewallProtoRule define ingress node firewall rule for TCP, UDP and SCTP protocols
 type IngressNodeFirewallProtoRule struct {
-	// +kubebuilder:validation:Optional
+	//Ports can be a single port or range of ports using start-end range format
+	//+kubebuilder:validation:Optional
 	// +optional
-	Port uint16 `json:"port"`
+	Ports string `json:"ports,omitempty"`
```

Comment:
Why not []uint16 ?

Reply:
I am using `start-end` range syntax and that can only be done as string right ?

Reply:
Ah true. Sorry for that, makes sense!